src/ZfSnapPhpDebugBar/Collector/DoctrineCollectorFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace ZfSnapPhpDebugBar\Collector;
+
+use Interop\Container\ContainerInterface;
+use DebugBar\Bridge\DoctrineCollector;
+use Doctrine\DBAL\Logging\DebugStack;
+use Zend\ServiceManager\FactoryInterface;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * Factory of `DoctrineCollector`.
+ * @author Vasilij Belosludcev <https://github.com/bupy7>
+ * @see DoctrineCollector
+ */
+class DoctrineCollectorFactory implements FactoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $debugStack = new DebugStack;
```

Comment:
Di is not enabled by default in zf3. Invokable factory was introduced in 2.7 https://github.com/zendframework/zend-servicemanager/releases/tag/release-2.7.0